src/libraries/System.Private.CoreLib/src/System/IO/FileStatus.Unix.cs

```diff
@@ -471,5 +531,8 @@ private static long UnixTimeSecondsToNanoseconds(DateTimeOffset time, long secon
             const long TicksPerSecond = TicksPerMillisecond * 1000;
             return (time.UtcDateTime.Ticks - DateTimeOffset.UnixEpoch.Ticks - seconds * TicksPerSecond) * NanosecondsPerTick;
         }
+
+        private static string GetHandlePath(SafeFileHandle handle)
+            => handle.Path ?? throw new ArgumentException(SR.Arg_HandleHasNoPath);
```

Comment:
A SafeFileHandle with a null path seems to be perfectly valid. The one you get back from `File.OpenHandle(..)` does set Path on it, and from `new SafeFileHandle(path..)`, but it need not. For example, you could have pinvoked to get a handle and wrapped it with SafeFileHandle.

So SetAccessOrWriteTimeCore should be able to operate with either a handle or a path. You even handle that:
```c#
            int rv = handle is not null ? Interop.Sys.FUTimens(handle, buf) :
                                          Interop.Sys.UTimensat(path, buf);
```
Please add tests for passing in SafeFileHandles that have a null path.

Reply:
@tmds I added a _Pathless Test class. Would you help me with the Unix-specific suggestions?

Reply:
@deeprobin what are the Unix suggestions you refer to? Maybe I can help.

Stepping back a moment. 

I want to respectfully repeat my request I made on a previous PR to set up WSL2 so you can run Linux tests locally.

Your goal before pushing is get the changes as ready as you can make them. This is because code reviews are really expensive for us (in this case 3 engineers have reviewed all or part, I have looked at this maybe 4 times). It also ties up a significant number of machines (at a wild guess 50-100 machine hours each push).

At the moment it looks like you pushed up new tests but not the matching change required to make the new tests pass on Linux (removing GetHandlePath() I guess)  

My request of you is to set up WSL2 so that you can run Linux tests locally. 

1. Follow online docs to enable WSL2 and install a Linux distro. Eg., Ubuntu 20.04. 
2. Follow workflow instructions in this repo, install prerequisites to build, clone your fork into this Linux install, and build.
3. Run tests in System.IO.FileSystem as described in the workflow instructions.

To share changes between Linux and Windows you can of course push to this PR but that would defeat the purpose as it would trigger validation again. Please don't do that. Instead you'll want to create a new branch and push/pull that. When you're happy and tests pass on both then you can push to the real branch so update this PR.

When you're comfortable with this, a common approach to make Linux work easier is to use VS Code. You can either install it on Ubuntu directly, or use it on Windows and connect to the WSL2 distro. It's super easy and seamless. The advantage of this is that you can make changes on Linux directly, test them, etc, and not have to return to Windows and push/pull for the smallest thing.

Reply:
> So SetAccessOrWriteTimeCore should be able to operate with either a handle or a path. You even handle that:

If we're using `fsetattrlist` as noted by @Jozkee in https://github.com/dotnet/runtime/pull/60507#discussion_r867062663, I think all APIs work even when the handle has no path.

The methods can then all have the signature with nullable handle and path: `SafeFileHandle? handle, string? path`.
Syscalls are made with the handle when it is not null, and with the path otherwise.

We still need a `string path` to use in error messages. Some literal can be used for handles with a `null` path.

```cs
        private static string GetErrorPath(string? path, SafeFileHandle? handle)
            => path ?? handle!.Path ?? "<<handle>>";
```

`File_GetSetTimes_SafeFileHandle_Pathless` tests should then all pass.

Reply:
@danmoseley Thanks again for the other tips. This was one of my first PRs so the quality was not the best at first.

WSL2 works for me now, but I'm not familiar with the Unix file system yet.

Also I don't know how to test the OSX specific stuff (see CI failures). I am not in ownership of an Apple device :D.

@tmds I think we need to define `fsetattrlist` somewhere. See CI Failures. How to do that?

Reply:
> Also I don't know how to test the OSX specific stuff (see CI failures). I am not in ownership of an Apple device :D.

That's fine. Neither do I at the moment. Testing on both Windows and Linux covers most things.